tests/utils/test_dict.py

```diff
@@ -181,4 +198,97 @@ def test_equal(self):
         }
 
         got = equal(d1, d2)
+        self.assertFalse(got)
+
+    def test_flatKeys(self):
+        d = {
+            'a': 22,
+            'b': {
+                'a': 'hey',
+                'b': {
+                    'a': 2,
+                },
+            },
+        }
+
+        got = flatKeys(d)
+        expected = [ 'a', 'b.a', 'b.b.a' ]
+
+        self.assertListEqual(got, expected)
+
+        d = {
+            'a': 22,
+            'b': [
+                { 'a': 'hi' },
+                { 'a': 'there' },
+                { 'b': 'friend' },
+            ],
+        }
+
+        got = flatKeys(d)
+        expected = [ 'a', 'b.[0].a', 'b.[1].a', 'b.[2].b' ]
+
+        self.assertListEqual(got, expected)
+
+    def test_subset(self):
+        d1 = {
+            'a': 22,
+        }
+
+        d2 = {
+            'a': 22,
+            'b': 23,
+        }
+
+        got = subset(d1, d2)
+        self.assertTrue(got)
+
+        got = subset(d2, d1)
+        self.assertFalse(got)
+
+        d1 = {
+            'a': 22,
+            'b': {
+                'a': 21,
+            },
+        }
+
+        d2 = {
+            'a': 22,
+            'b': {
+                'a': 21,
+                'b': 'hey',
+            },
+        }
+
+        got = subset(d1, d2)
+        self.assertTrue(got)
+
+        got = subset(d2, d1)
+        self.assertFalse(got)
+
+    def test_partialEqual(self):
```

Comment:
```
def partialEqual(d1, d2):
  return True
```
also passes all of these tests. I wonder if these tests are truly getting to the heart of the expected behavior of this method?